gm4_zauber_cauldrons/data/gm4/advancements/zauber_cauldrons_drink_all_magicol.json

```diff
@@ -0,0 +1,200 @@
+{
+    "display": {
+        "icon": {
+            "item": "minecraft:potion",
+            "nbt": "{CustomModelData:3420000}"
+        },
+        "title": {
+            "translate": "advancement.gm4.zauber_cauldrons.drink_all_magicol.title",
+            "fallback": "Taste the Rainbow"
+        },
+        "description": {
+            "translate": "advancement.gm4.zauber_cauldrons.drink_all_magicol.description",
+            "fallback": "Drink all colors of Magicol",
+            "color": "gray"
+        },
+        "frame": "challenge",
+        "hidden": false
+    },
+    "parent": "gm4:zauber_cauldrons_make_magicol",
+    "criteria": {
+        "white_magicol": {
```

Comment:
Not sure if we want to, but with mecha `v0.86.0` we can also use bolt in json files so these criteria could also be generated.

Reply:
That'd be preferrable, however, updating to mecha `v0.86.0` requires updating to beet `v0.100.0`, which in turn requires an update to pydantic `v2` (from `v1`).

I'm not entirely sure how much that pydantic update will break, hence I'd suggest to do this update in a separate PR instead of this one.

Reply:
I believe #918 updates beet so yeah, good idea to leave it here

Reply:
#915 does indeed handle this upgrade. Beet doesn't fully support v2 yet so there are no changes to be made yet (just replacing the imports), but I have an incomplete branch with the start of some upgrades when eventually we do the upgrade.